corehq/motech/repeaters/tasks.py

```diff
@@ -215,6 +226,210 @@ def _process_repeat_record(repeat_record):
         )
 
 
+@periodic_task(
+    run_every=PROCESS_REPEATERS_INTERVAL,
+    queue=settings.CELERY_PERIODIC_QUEUE,
+)
+def process_repeaters():
+    """
+    Processes repeaters, instead of processing repeat records
+    independently the way that ``check_repeaters()`` does.
+    """
+    process_repeater_lock = get_redis_lock(
+        PROCESS_REPEATERS_KEY,
+        timeout=24 * 60 * 60,
+        name=PROCESS_REPEATERS_KEY,
+    )
+    if not process_repeater_lock.acquire(blocking=False):
+        return
+    try:
+        for domain, repeater_id, lock_token in iter_ready_repeater_ids_forever():
+            process_repeater.delay(domain, repeater_id, lock_token)
```

Comment:
> Do we need to be checking how long this task has been running for and exit if we are about to exceed the `process_repeater_lock` timeout value?

`process_repeater()` should take a maximum of five minutes, if that repeater's requests are timing out. The lock timeout is three hours.

I expect we would only exceed the timeout if Celery was killed and the `update_repeater()` task was never called. So the timeout is intended to prevent a repeater from being locked forever.

I think it could be useful to use Datadog to monitor how long the task takes, but probably only to identify problematic endpoints, not to exit the task.

Reply:
I may not be understanding, but I'm referring to the `process_repeaters` task and the `process_repeater_lock` with the 24 hour timeout.

Reply:
Sorry for my confusion. I didn't read your question properly.

I wondered about that timeout too, because there is no problem if this loop just keeps going for more than 24 hours _as long as_ new repeat records are being sent in a reasonable time.

What do you think of not having a timeout?

Reply:
4c4189661f8